Test failure: Robot kinematics not updating position

## Test Failure Description

**Test:** `TestDifferentialDriveRobot.test_kinematics_update`
**File:** `robocar/simulation/tests/test_robot_model.py:186`

### Error Details
```
assert robot.state.x > initial_x
AssertionError: assert 0.0 > 0.0

tests/test_robot_model.py:186: AssertionError
```

### Expected Behavior
When both motors are set to PWM 100 for forward motion and the robot is updated for 100 iterations (1 second), the robot should move forward and its x position should increase.

### Actual Behavior
The robot's position remains at (0.0, 0.0) despite setting motor commands and running the simulation loop.

### Test Configuration
- Both motors set to PWM 100
- 100 simulation iterations at 0.01s timestep
- Expected forward motion along x-axis

### Root Cause
The kinematic update mechanism in `DifferentialDriveRobot` is not properly converting motor commands to position changes. This could be related to:
1. PWM to velocity conversion not working
2. Motor RPM not being updated from PWM commands  
3. Kinematic equations not properly integrating velocity to position

### Dependencies
This issue may be related to issue #2 (DCMotor dynamics) as the robot kinematics depend on proper motor modeling.

### Location
`robocar/simulation/src/robot_model.py` - DifferentialDriveRobot class

--- a/.clang-format
+++ b/.clang-format
@@ -34,7 +34,7 @@ DerivePointerAlignment: false
 BreakBeforeBraces: Linux
 
 # Other Breaking
-AlwaysBreakAfterReturnType: None
+BreakAfterReturnType: Automatic
 AlwaysBreakBeforeMultilineStrings: false
 BreakBeforeBinaryOperators: None
 BreakBeforeTernaryOperators: true

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -15,3 +15,15 @@ updates:
     labels:
       - "dependencies"
       - "github-actions"
+
+  # Python dependencies (robocar-simulation)
+  - package-ecosystem: "pip"
+    directory: "/packages/esp32-projects/robocar-simulation"
+    schedule:
+      interval: "weekly"
+    commit-message:
+      prefix: "chore"
+      include: "scope"
+    labels:
+      - "dependencies"
+      - "python"

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -18,7 +18,16 @@ mcu-tinkering-lab/
 │       ├── robocar-docs/           # Documentation and coordination justfile
 │       ├── esp32cam-llm-telegram/  # Telegram bot with LLM vision
 │       ├── esp32-cam-webserver/    # Live video streaming server
-│       └── esp32-cam-i2s-audio/    # Camera + audio processing
+│       ├── esp32-cam-i2s-audio/    # Camera + audio processing
+│       ├── gamepad-synth/          # PS4 gamepad-controlled I2S audio synthesizer
+│       ├── kids-audio-toy/         # Potentiometer-controlled audio toy
+│       ├── audiobook-player/       # ESPHome audiobook player
+│       ├── xbox-switch-bridge/     # Xbox BLE to Switch USB bridge
+│       ├── switch-usb-proxy/       # Switch USB protocol proxy
+│       ├── it-troubleshooter/      # IT troubleshooting assistant
+│       ├── nfc-scavenger-hunt/     # NFC-based scavenger hunt game
+│       ├── esp32-wifitest/         # WiFi AP test firmware
+│       └── esp32-wireguard-ha-example/ # WireGuard + Home Assistant (ESPHome)
 ├── .github/workflows/              # CI/CD (6 workflows)
 ├── tools/scaffold/                 # Project scaffolding scripts
 ├── docs/blueprint/                 # Architecture docs (PRDs, ADRs)
@@ -40,7 +49,7 @@ mcu-tinkering-lab/
 | Python type checker | mypy |
 | Secret scanning | gitleaks |
 | Pre-commit | pre-commit (8 hooks) |
-| CI/CD | GitHub Actions (6 workflows) |
+| CI/CD | GitHub Actions (11+ workflows) |
 | Containers | Docker + docker-compose |
 
 ## Build Commands

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -35,17 +35,17 @@ git clone https://github.com/laurigates/mcu-tinkering-lab.git
 cd mcu-tinkering-lab
 
 # Build Docker images
-make docker-build
+just docker-build
 
 # Start development shell
-make docker-dev
+just docker-dev
 ```
 
 #### Option 2: Native Setup
 
 ```bash
 # Install development tools
-make install-dev-tools
+just install-dev-tools
 
 # This will install:
 # - pre-commit hooks
@@ -93,19 +93,19 @@ git checkout -b fix/bug-description
 
 ```bash
 # Format code
-make format
+just format
 
 # Run linters
-make lint
+just lint
 
 # Check formatting (non-destructive)
-make format-check
+just format-check
 
 # Build affected projects
-make build-all
+just build-all
 
 # Run tests (when available)
-make test-all
+just test-all
 ```
 
 ### 4. Commit Your Changes
@@ -138,10 +138,10 @@ We use **clang-format** with Google style (4-space indent, 100 column limit).
 **Formatting:**
 ```bash
 # Format all C/C++ files
-make format-c
+just format-c
 
 # Check formatting without modifying
-make format-check-c
+just format-check-c
 ```
 
 **Style Rules:**
@@ -185,10 +185,10 @@ We use **ruff** for linting and formatting (PEP 8 compatible, 100 character line
 **Formatting:**
 ```bash
 # Format all Python files
-make format-python
+just format-python
 
 # Lint Python code
-make lint-python
+just lint-python
 ```
 
 **Style Rules:**
@@ -345,8 +345,8 @@ Optional scope to specify which part of the codebase is affected:
 ### 1. Before Creating PR
 
 - ✅ All tests pass
-- ✅ Code is formatted (`make format`)
-- ✅ Linters pass (`make lint`)
+- ✅ Code is formatted (`just format`)
+- ✅ Linters pass (`just lint`)
 - ✅ Documentation updated
 - ✅ Commit messages follow convention
 - ✅ Branch is up-to-date with main

--- a/README.md
+++ b/README.md
@@ -78,7 +78,16 @@ mcu-tinkering-lab/
 │   │   ├── robocar-docs/           # 📚 Documentation and coordination justfile
 │   │   ├── esp32cam-llm-telegram/  # 💬 Telegram bot with LLM vision
 │   │   ├── esp32-cam-webserver/    # 🌐 Live video streaming server
-│   │   └── esp32-cam-i2s-audio/    # 🔊 Camera + audio processing
+│   │   ├── esp32-cam-i2s-audio/    # 🔊 Camera + audio processing
+│   │   ├── gamepad-synth/          # 🎵 PS4 gamepad I2S audio synthesizer
+│   │   ├── kids-audio-toy/         # 🧸 Potentiometer-controlled audio toy
+│   │   ├── audiobook-player/       # 📖 ESPHome audiobook player
+│   │   ├── xbox-switch-bridge/     # 🎮 Xbox BLE to Switch USB bridge
+│   │   ├── switch-usb-proxy/       # 🔌 Switch USB protocol proxy
+│   │   ├── it-troubleshooter/      # 🔧 IT troubleshooting assistant
+│   │   ├── nfc-scavenger-hunt/     # 📱 NFC-based scavenger hunt game
+│   │   ├── esp32-wifitest/         # 📡 WiFi AP test firmware
+│   │   └── esp32-wireguard-ha-example/ # 🔒 WireGuard + Home Assistant
 │   ├── arduino-projects/           # 🚧 Coming soon
 │   ├── stm32-projects/             # 🚧 Coming soon
 │   └── shared-libs/                # 📦 Shared code libraries (planned)
@@ -351,7 +360,7 @@ just docker-clean
 
 | Platform | Status | Projects | Tests | CI/CD |
 |----------|--------|----------|-------|-------|
-| **ESP32** | ✅ Active | 7 projects | 🚧 In progress | ✅ Automated |
+| **ESP32** | ✅ Active | 16 projects | 🚧 In progress | ✅ Automated |
 | **Arduino** | 🚧 Planned | 0 | ❌ N/A | ❌ N/A |
 | **STM32** | 🚧 Planned | 0 | ❌ N/A | ❌ N/A |
 | **Simulation** | ✅ Active | Python 3.11 | ✅ pytest | ✅ Automated |

--- a/docs/blueprint/README.md
+++ b/docs/blueprint/README.md
@@ -47,6 +47,7 @@ docs/blueprint/
 - PRD-005: Xbox-to-Switch Bridge
 - PRD-006: NFC Scavenger Hunt
 - PRD-007: Audiobook Player
+- PRD-008: Gamepad Synth
 
 ### ADRs (Architecture Decision Records)
 - ADR-001: Monorepo Structure for Multi-Platform MCU Projects
@@ -59,6 +60,8 @@ docs/blueprint/
 - ADR-008: Gemini Robotics-ER 1.5 as Third AI Backend
 - ADR-009: Switch Pro Controller On-Device Protocol
 - ADR-010: MQTT Logging Architecture
+- ADR-011: Gamepad Synth I2S Audio
+- ADR-012: Browser-Based Web Flasher Architecture
 
 ### PRPs (Product Requirement Prompts)
 Located in `docs/prps/`:

--- a/docs/blueprint/adrs/ADR-012-web-flasher-architecture.md
+++ b/docs/blueprint/adrs/ADR-012-web-flasher-architecture.md
@@ -1,4 +1,4 @@
-# ADR-007: Browser-Based Web Flasher Architecture
+# ADR-012: Browser-Based Web Flasher Architecture
 
 ## Status
 

--- a/docs/blueprint/manifest.md
+++ b/docs/blueprint/manifest.md
@@ -16,7 +16,17 @@ Primary project: dual-ESP32 AI-powered robot car with pluggable AI vision backen
 | ADR-004 | ADR | OTA Firmware Update Architecture | accepted | 2026-03-09 |
 | ADR-005 | ADR | IT Troubleshooter Hardware Selection | accepted | 2026-03-12 |
 | ADR-006 | ADR | USB Composite Device Architecture | accepted | 2026-03-15 |
-| ADR-007 | ADR | Browser-Based Web Flasher Architecture | accepted | 2026-04-03 |
+| ADR-007 | ADR | Shared I2C Protocol Component | accepted | 2026-03-12 |
+| ADR-008 | ADR | Gemini Robotics-ER 1.5 as Third AI Backend | draft | 2026-03-18 |
+| ADR-009 | ADR | Switch Pro Controller On-Device Protocol | accepted | 2026-03-20 |
+| ADR-010 | ADR | MQTT Logging Architecture | accepted | 2026-03-25 |
+| ADR-011 | ADR | Gamepad Synth I2S Audio | accepted | 2026-04-07 |
+| ADR-012 | ADR | Browser-Based Web Flasher Architecture | accepted | 2026-04-03 |
+| PRD-004 | PRD | IT Troubleshooter | draft | 2026-03-12 |
+| PRD-005 | PRD | Xbox-to-Switch Bridge | draft | 2026-03-15 |
+| PRD-006 | PRD | NFC Scavenger Hunt | draft | 2026-03-18 |
+| PRD-007 | PRD | Audiobook Player | draft | 2026-03-22 |
+| PRD-008 | PRD | Gamepad Synth | draft | 2026-04-07 |
 
 ## Project Overview
 
@@ -26,6 +36,6 @@ Primary project: dual-ESP32 AI-powered robot car with pluggable AI vision backen
 | Primary Language (Firmware) | C/C++ (ESP-IDF v5.4+) |
 | Primary Language (Simulation) | Python 3.11 |
 | Build System | CMake via ESP-IDF, Make, justfile |
-| CI/CD | GitHub Actions (6 workflows) |
-| Active Platforms | ESP32 (7 projects), Simulation |
+| CI/CD | GitHub Actions (11+ workflows) |
+| Active Platforms | ESP32 (16 projects), Simulation |
 | Planned Platforms | Arduino, STM32 |

--- a/docs/health-history.json
+++ b/docs/health-history.json
@@ -27,6 +27,20 @@
       },
       "fixed_count": 0,
       "remaining_count": 3
+    },
+    {
+      "date": "2026-04-08",
+      "overall_score": 86,
+      "grade": "B",
+      "category_scores": {
+        "docs": 18,
+        "tests": 10,
+        "security": 18,
+        "quality": 16,
+        "ci": 20
+      },
+      "fixed_count": 10,
+      "remaining_count": 4
     }
   ]
 }

--- a/justfile
+++ b/justfile
@@ -76,7 +76,7 @@ install-dev-tools:
     #!/usr/bin/env bash
     set -euo pipefail
     pip install --upgrade pip
-    pip install pre-commit==3.8.0 ruff==0.6.9 mypy==1.11.2 pytest==8.3.3 pytest-cov==5.0.0 uv==0.4.18
+    pip install pre-commit ruff mypy pytest pytest-cov uv
     pre-commit install
     command -v clang-format >/dev/null 2>&1 || echo "clang-format not found — install with: brew install clang-format"
     command -v cppcheck >/dev/null 2>&1 || echo "cppcheck not found — install with: brew install cppcheck"
@@ -169,7 +169,7 @@ lint-c:
             --suppress=unmatchedSuppression \
             --suppress=unusedStructMember \
             --inline-suppr \
-            --error-exitcode=0 \
+            --error-exitcode=1 \
             --template=gcc \
             2>&1 | tee tmp/cppcheck-report.txt
     echo "C/C++ lint checks passed"
@@ -180,7 +180,7 @@ lint-python:
     #!/usr/bin/env bash
     set -euo pipefail
     command -v ruff >/dev/null 2>&1 || { echo "Warning: ruff not found — pip install ruff"; exit 0; }
-    cd {{robocar_sim_dir}} && ruff check .
+    ruff check .
     echo "Python lint checks passed"
 
 # Format all code (C/C++ and Python)
@@ -211,7 +211,7 @@ format-python:
     #!/usr/bin/env bash
     set -euo pipefail
     command -v ruff >/dev/null 2>&1 || { echo "Warning: ruff not found"; exit 0; }
-    cd {{robocar_sim_dir}} && ruff format .
+    ruff format .
     echo "Python code formatted"
 
 # Check formatting without modifying files
@@ -242,7 +242,7 @@ format-check-python:
     #!/usr/bin/env bash
     set -euo pipefail
     command -v ruff >/dev/null 2>&1 || { echo "Warning: ruff not found"; exit 0; }
-    cd {{robocar_sim_dir}} && ruff format --check .
+    ruff format --check .
     echo "Python formatting check passed"
 
 # Build robocar projects (use `just <module>::build` for other projects)

--- a/mypy.ini
+++ b/mypy.ini
@@ -1,0 +1,26 @@
+# Mypy configuration for MCU Tinkering Lab
+# The robocar-simulation subproject has stricter settings in its own pyproject.toml.
+# This root config provides sensible defaults for tools/ and other Python files.
+
+[mypy]
+python_version = 3.11
+warn_return_any = True
+warn_unused_configs = True
+check_untyped_defs = True
+no_implicit_optional = True
+
+# Third-party modules used by tools/ that lack type stubs
+[mypy-objc.*]
+ignore_missing_imports = True
+
+[mypy-CoreWLAN.*]
+ignore_missing_imports = True
+
+[mypy-hid.*]
+ignore_missing_imports = True
+
+[mypy-serial.*]
+ignore_missing_imports = True
+
+[mypy-pymunk.*]
+ignore_missing_imports = True

